shark/iree_utils/compile_utils.py

```diff
@@ -68,7 +68,13 @@ def compile_module_to_flatbuffer(
     # Setup Compile arguments wrt to frontends.
     input_type = ""
     args = get_iree_frontend_args(frontend)
-    args += get_iree_device_args(device)
+    if device in [
```

Comment:
There can be more args appended in vulkan_device settings, you should bypass the target triple search rather than bypassing the whole get_iree_device_args function.

Reply:
percolated extra_args all the way to vulkan utils
Can have a more structured way to handle it later